### DISCUSSION TITLE
From Docker compose to APPUiO cloud

### DISCUSSION THREAD

**Discussion by dbartenstein** [appuio/appuio-cloud-community#43] [Help Needed]
Hello!

Our Django application is running with Docker compose on our local development computers (On the servers the application is currently run on Debian servers - without docker nor kubernetes).

In order to deploy the Django app on APPUiO cloud, what does a good preparation look like? We thought about getting our Django application up and running with [minikube](https://minikube.sigs.k8s.io/docs/) on our local development machines first. What do you think of this approach?

Thanks a lot for your recommendations!

**Comment by tobru** [comment]
For running a Python application on APPUiO Cloud we have a [getting started guide](https://docs.appuio.cloud/user/tutorials/getting-started/python-web.html) available. 

As for testing locally: minikube is based on plain Kubernetes, as APPUiO Cloud is based on OpenShift 4, you might miss some specialities of OpenShift, mainly regarding permissions. For running OpenShift 4 locally, there is [CodeReady Containers](https://developers.redhat.com/products/codeready-containers/overview) available, but needs quite some resources. As APPUiO Cloud is billed on the usage, you could also already start with APPUiO Cloud to make your deployment ready. When you're not working on it, just scale it down so that no Pod runs, this will then not cause any costs. And if your application doesn't use that much RAM, it's quite [cheap](https://products.docs.vshn.ch/products/appuio/cloud/pricing.html#_price_calculator).

**Reply by dbartenstein** [reply]
@tobru thanks for your reply and the link to the getting started guide!

In general we’d like to not being "locked in" and use Kubernetes for local development for now. We’d like to keep our options open to running our app with e.g. Rancher.
Having the application running fine in minikube (instead of docker-compose) would probably ease deploying our app to APPUiO - is this assumption valid? If yes, that would be one of the first steps we’d take.

Also thank your for hinting on using APPUiO cloud to making the deployment ready.

**Reply by tobru** [reply]
> Having the application running fine in minikube (instead of docker-compose) would probably ease deploying our app to APPUiO - is this assumption valid?

In 99% of all cases this works, if your image follows some best-practices. See [Creating images](https://docs.openshift.com/container-platform/4.10/openshift_images/create-images.html#images-create-guide-openshift_create-images), especially "Support arbitrary user ids".

In addition to minikube, I suggest you check [Rancher Desktop](https://rancherdesktop.io/), which is a very developer-friendly local Kubernetes environment.
